### PR TITLE
Fixed listItem primaryText, secondaryText and adminBadge being off center

### PR DESCRIPTION
**Pull request by Ralphvandodewaard**

### CODE REVIEW

File: src/admin-badge/styles.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const styleSheet = {
     backgroundColor: colors.theme,
     color: colors.white,
     fontSize: '10px',
-    lineHeight: '10px',
+    lineHeight: '23px',
```
Comment (IanCStewart) [inline src/admin-badge/styles.js]: This doesn't seem logical. And it looks kind of off aswell. Please revert this

File: src/list-item/styles.js [hunk 2]
```diff
@@ -70,8 +70,8 @@ export default {
     fontSize: '14px',
     fontWeight: 'normal',
     marginBottom: 0,
-    marginTop: '2.4px',
-    lineHeight: '22px',
+    marginTop: '2px',
+    lineHeight: '19px',
```
Comment (IanCStewart) [inline src/list-item/styles.js]: Please only use the `marginTop` to get it "centered". `LineHeight` should stay the same as the fontSize

**Comment by Ralphvandodewaard** [comment]
In this commit, overflow of `secondaryText` will be hidden, but I wasn't able to add `textOverflow: 'ellipsis'`. This has to do with a problem regarding `overflow-x` and `overflow-y` which also occured in [this](https://github.com/anchorchat/anchor-ui/pull/475) issue.

Sources:
https://css-tricks.com/popping-hidden-overflow/
http://stackoverflow.com/questions/6421966/css-overflow-x-visible-and-overflow-y-hidden-causing-scrollbar-issue

File: src/list-item/styles.js [hunk 3]
```diff
@@ -17,6 +17,7 @@ export default {
     paddingLeft: '10px',
     paddingBottom: '10px',
     position: 'relative',
+    overflow: 'hidden',
```
Comment (IanCStewart) [inline src/list-item/styles.js]: I don't understand why we need `overflow: hidden` here. Could u explain this?

Comment (Ralphvandodewaard) [inline src/list-item/styles.js]: As explained in the sources above, it is not possible to use both `overflow-x: hidden` and `overflow-y: visible` on the same element. For the `AdminBadge` to be fully visible, the `overflow-y` must be set to `visible` and thus the `overflow-x` cannot be `hidden` which causes long texts in `secondaryText` to overflow. In the sources above it is then also explained that this issue can be worked around by adding `overflow: hidden` to the closest positioned ancestor, which in this case is the root of `ListItem`. By doing this, long texts in `secondaryText` will not overflow and the `AdminBadge` will be fully visible.

**Comment by Ralphvandodewaard** [comment]
Explanation on changes made:

- `overflow: hidden` has been added to `root` to prevent long texts in `secondaryText` from overflowing (source: https://css-tricks.com/popping-hidden-overflow/)

- The `lineHeight` of both `primaryText` and `secondaryText` have been changed to match the values of their `fontSize` (as Ian requested)

- `paddingBottom: 1px` has been added to `primaryText` to prevent descender letters (i.e. g, j, y) from being cut off by the `overflow: hidden`

- `paddingTop: 2px` has been added to `secondaryText` to add whitespace between `primaryText` and `secondaryText` while keeping both centered

File: docs/components.json [hunk 4]
```diff
@@ -2673,6 +2673,17 @@
           "computed": false
         }
       },
+      "secondaryBadge": {
```
Comment (sjaakluthart) [inline docs/components.json]: I don't like this prop name, it implies it has something to do with `badge`, please change it to something like textBadge with this prop description `A badge to display before the secondaryText.`

File: src/list-item/get-styles.js [hunk 5]
```diff
@@ -32,13 +32,17 @@ function root(themeColor, active, rightButton, avatar, overrideStyle) {
   return combineStyles(style, overrideStyle);
 }
 
-function text(textStyle, active, overrideStyle) {
+function text(textStyle, active, secondaryBadge, overrideStyle) {
```
Comment (sjaakluthart) [inline src/list-item/get-styles.js]: Also change this argument name.

**Pull request merged**